crates/iceberg-catalog/src/implementations/postgres/task_queues.rs

```diff
@@ -0,0 +1,1114 @@
+use chrono::Utc;
+use iceberg_ext::catalog::rest::IcebergErrorResponse;
+use itertools::Itertools;
+use sqlx::{PgConnection, PgPool};
+use uuid::Uuid;
+
+use crate::{
+    implementations::postgres::{dbutils::DBErrorHandler, ReadWrite},
+    service::task_queue::{Task, TaskFilter, TaskQueue, TaskQueueConfig, TaskStatus},
+};
+
+#[derive(Debug, Clone)]
+pub struct PgQueue {
+    pub read_write: ReadWrite,
+    pub config: TaskQueueConfig,
+    pub max_age: sqlx::postgres::types::PgInterval,
+}
+
+impl PgQueue {
+    /// Creates a new `PgQueue` instance from the provided `ReadWrite` and `TaskQueueConfig`.
+    ///
+    /// # Errors
+    /// Returns an error if the `max_age` duration in the `TaskQueueConfig` is invalid.
+    pub fn from_config(read_write: ReadWrite, config: TaskQueueConfig) -> anyhow::Result<Self> {
+        let microseconds = config
+            .max_age
+            .num_microseconds()
+            .ok_or(anyhow::anyhow!("Invalid max age duration for task queues."))?;
+        Ok(Self {
+            read_write,
+            config,
+            max_age: sqlx::postgres::types::PgInterval {
+                months: 0,
+                days: 0,
+                microseconds,
+            },
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl TaskQueue for PgQueue {
+    async fn pick_new_task(&self, queue_name: &str) -> crate::api::Result<Option<Task>> {
+        pick_task(&self.read_write.write_pool, queue_name, &self.max_age).await
+    }
+
+    async fn record_success(&self, id: Uuid, message: Option<&str>) -> crate::api::Result<()> {
```

Comment:
Lets name the arguments `task_id` instead of just id to avoid confusion with entity_id.
Should we introduce a `TaskId` type?